gpt/git-tools.js

```diff
@@ -0,0 +1,95 @@
+// File: imports.js
+const { Tool } = require("langchain/tools");
+const execSync = require("child_process").execSync;
+const z = require('zod');
+const { LLMSingleActionAgent, AgentActionOutputParser, AgentExecutor } = require("langchain/agents");
+const { LLMChain } = require("langchain/chains");
+const { OpenAI, OpenAIChat } = require("langchain/llms/openai");
+const { BasePromptTemplate, BaseStringPromptTemplate, SerializedBasePromptTemplate,renderTemplate } = require("langchain/prompts");
+const {  InputValues, PartialValues, AgentStep, AgentAction, AgentFinish } = require("langchain/schema");
+const { Calculator } = require("langchain/tools/calculator");
+const { BufferWindowMemory } = require("langchain/memory");
+const path = require('path');
+const {GitCommitOutputParser} = require('./outputs/GitCommitOutputParser');
+const { formatInstructions, SUFFIX, PREFIX } = require('./git/instructions');
+const { GitTool, GitDiffStagedTool, GitStagedFilesTool, GitDiffTool, GitStatusTool, GitCommitTool, GitAddTool, GitNewBranchTool, GitCheckoutBranchTool, GitPullTool, GitPushTool, GitBranchListTool, GitDeleteBranchTool } = require('./git/tools');
+
+// File: GitCommitPromptTemplate.js
+class GitCommitPromptTemplate extends BaseStringPromptTemplate {
+  constructor(args) {
+    super({ inputVariables: args.inputVariables });
+    this.tools = args.tools;
+  }
+  _getPromptType() {
+    return 'prompt';
+  }
+  async processFile(file) {
+    const fileName = file.match(/(?<=diff --git a\/).*?(?= b\/)/);
+    if(!fileName?.[0]) {
+      return '';
+    }
```

Comment:
The GitTool class and its subclasses are tightly coupled with the execSync function from the child_process module. Consider abstracting the execSync function into a separate module or class to make your code more testable and modular.